src/action/mod.rs

```diff
@@ -407,33 +407,28 @@ impl Action {
 
     /// Run the requested action.
     async fn run_check(self, documents: Documentation, config: Config) -> Result<Finish> {
-        let n_cpus = num_cpus::get();
-
         let checkers = Checkers::new(config)?;
-
-        // TODO per file clustering might make sense here
-        let mistakes_count = stream::iter(documents.iter().enumerate())
-            .map(move |(idx, (origin, chunks))| {
-                let suggestions = checkers.check(origin, &chunks[..]);
-                async move { Ok::<_, color_eyre::eyre::Report>((idx, origin, suggestions?)) }
-            })
-            .buffered(n_cpus)
-            .try_fold(0_usize, |acc, (_idx, origin, suggestions)| async move {
-                let n = suggestions.len();
-                let path = origin.as_path();
-                if n == 0 {
-                    log::info!("✅ {}", path.display());
-                } else {
-                    log::info!("❌ {} : {}", path.display(), n);
-                }
-                for suggestion in suggestions {
-                    println!("{}", suggestion);
-                }
-                Ok::<_, color_eyre::eyre::Report>(acc + n)
+        let num_mistakes = documents
+            .into_par_iter()
+            .map(|(origin, chunks)| {
+                checkers.check(&origin, &chunks).map(|suggestions| {
+                    let path = origin.as_path();
+                    let n = suggestions.len();
+                    match suggestions.is_empty() {
+                        true => log::info!("✅ {}", path.display()),
+                        false => log::info!("❌ {} : {}", path.display(), n),
+                    };
+                    for suggestion in suggestions {
+                        println!("{}", suggestion);
+                    }
```

Comment:
Wouldn't the log output be scrambled/interleaved with one another?

Reply:
It doesn't practically.. :monocle_face: that is a bit fishy

Reply:
This is a fair concern - we should probably fan the work out to threads, and only present on the main thread